### ISSUE TITLE
Rule Proposal: no-spaced-func with never

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6080]
From [requireSpacesInCallExpression] and [disallowSpacesInCallExpression].

In JSCS, we can specify that spaces are required before the open parenthesis of call expressions.
In ESLint, [no-spaced-func] has a responsibility for spacing of call expressions.

This proposal is to add `"always"`/`"never"` option to [no-spaced-func].
Maybe we should rename this rule to `call-spacing` or something like.

```js
{
    "call-spacing": ["error", "always" or "never"]
}
```

- `"always"` - Requires spacing before the open parenthesis of call expressions.
- `"never"` (default) - Disallows spacing before the open parenthesis of call expressions. (the current behavior)

[requireSpacesInCallExpression]: http://jscs.info/rule/requireSpacesInCallExpression
[disallowSpacesInCallExpression]: http://jscs.info/rule/disallowSpacesInCallExpression
[no-spaced-func]: http://eslint.org/docs/rules/no-spaced-func

**Comment by alberto** [comment]
I can champion this one

**Comment by platinumazure** [comment]
I would support a rename. Double negatives would be confusing.

**Comment by alberto** [comment]
Good point, I agree. Maybe `func-spacing` or `func-parens-spacing`?

**Comment by platinumazure** [comment]
I liked the `"call-spacing"` @mysticatea had proposed. (Sorry, when I said "support" I was referring to his proposal, but I should have made that more clear.)

**Comment by mysticatea** [comment]
I thought that `func-spacing` is confusing between call expressions and function declarations. It's the reason I chose `call` word.

**Comment by alberto** [comment]
Yeah, sorry, I lost focus on the scope.

**Comment by nzakas** [comment]
How about `func-call-spacing`, as I'm not sure "call" is obvious on its own?

**Comment by nzakas** [comment]
Per TSC meeting (7/21/16), marking as accepted.

Note that is proposal includes deprecating `no-spaced-func` in favor of the new `func-call-spacing` rule.

**Comment by mysticatea** [comment]
Should we need to tweak `eslint:all` preset for deprecated rules?

**Comment by alberto** [comment]
@mysticatea I think it's better to discuss that as a separate issue

**Comment by mysticatea** [comment]
I opened #6734.

**Comment by btmills** [comment]
Working on this.
